### PR TITLE
Relabel pod and service dimensions for non-pod metrics

### PR DESCRIPTION
**Pull request by sumo-drosiek**
###### Description

Relabel pod and service dimensions for non-pod metrics

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/helm/prometheus-overrides.yaml [hunk 1]
```diff
@@ -193,6 +193,21 @@ prometheus:
...
+      ## kube_statefulset_status_observed_generation
+      ## kube_statefulset_status_replicas
+      - url: http://$(CHART).$(NAMESPACE).svc.cluster.local:9888/prometheus.metrics.state
+        writeRelabelConfigs:
+          - action: keep
+            regex: kube-state-metrics;(?:kube_statefulset_status_observed_generation|kube_statefulset_status_replicas|kube_statefulset_replicas|kube_statefulset_metadata_generation|kube_daemonset_status_current_number_scheduled|kube_daemonset_status_desired_number_scheduled|kube_daemonset_status_number_misscheduled|kube_daemonset_status_number_unavailable|kube_deployment_spec_replicas|kube_deployment_status_replicas_available|kube_deployment_status_replicas_unavailable|kube_node_info|kube_node_status_allocatable|kube_node_status_capacity|kube_node_status_condition)
+            sourceLabels: [job, __name__]
+          - action: labelmap
+            regex: (pod|service)
+            replacement: sumo_metrics_${1}
```
Comment (frankreno) [inline deploy/helm/prometheus-overrides.yaml:207]: Nit.  sumo_metrics_ as the prefix feels unintuitive. In this case, `pod` and `service` are pointers to the pod/service that Prometheus scraped the data from.  Let's change it `sd_` to indicate it is part of the service discovery mechanism.

Comment (sumo-drosiek) [reply deploy/helm/prometheus-overrides.yaml]: Fixed

**Review by pmalek-sumo — APPROVED** [review]
🚢

File: deploy/helm/prometheus-overrides.yaml (hunk 1 above)
Comment (perk) [reply deploy/helm/prometheus-overrides.yaml]: I think we should go further and use `service_discovery_${1}` - no need to use the non-obvious abbreviation.

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -970,6 +970,21 @@ prometheus-operator:
...
+        ## kube_statefulset_status_observed_generation
+        ## kube_statefulset_status_replicas
+        - url: http://$(CHART).$(NAMESPACE).svc.cluster.local:9888/prometheus.metrics.state
+          writeRelabelConfigs:
+            - action: keep
+              regex: kube-state-metrics;(?:kube_statefulset_status_observed_generation|kube_statefulset_status_replicas|kube_statefulset_replicas|kube_statefulset_metadata_generation|kube_daemonset_status_current_number_scheduled|kube_daemonset_status_desired_number_scheduled|kube_daemonset_status_number_misscheduled|kube_daemonset_status_number_unavailable|kube_deployment_spec_replicas|kube_deployment_status_replicas_available|kube_deployment_status_replicas_unavailable|kube_node_info|kube_node_status_allocatable|kube_node_status_capacity|kube_node_status_condition)
+              sourceLabels: [job, __name__]
+            - action: labelmap
+              regex: (pod|service)
+              replacement: sd_${1}
```
Comment (perk) [inline deploy/helm/sumologic/values.yaml:984]: I think we should go further and use `service_discovery_` - no need to use the non-obvious `sd_` abbreviation.
```suggestion
              replacement: service_discovery_${1}
```

**Review by samjsong — APPROVED** [review]
LGTM, but +1 to Perk's suggestion

**Pull request merged**